silverback/middlewares.py

```diff
@@ -56,7 +56,7 @@ def _create_label(self, message: TaskiqMessage) -> str:
             return message.task_name
 
     def pre_execute(self, message: TaskiqMessage) -> TaskiqMessage:
-        if not (task_type := message.labels.pop("task_type")):
+        if "task_type" not in message.labels or not (task_type := message.labels.pop("task_type")):
```

Comment:
Should do the same thing but also clearer

Reply:
Not sure what you mean

Reply:
If you mean it should work as it was originally, that's not the case.

```python
>>> x = {"a": 1}
>>> x.pop('b')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'b'
```

Reply:
`.pop` returns None if the key is not in the dictionary,  which is falsey, so `not (var := None)` evals to True

Reply:
> If you mean it should work as it was originally, that's not the case.
> 
> ```python
> >>> x = {"a": 1}
> >>> x.pop('b')
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
> KeyError: 'b'
> ```

Ohbyes you're right, I was thinking about `.get`, good catch

Reply:
Could also do `.pop("key", None)` to get that same effect

Reply:
This is probably the most clear and concise:
```py
if "task_type" not in message.labels:
    return message

task_type = message.labels.pop("task_type")
```